context/app/static/js/pages/Workspace/Workspace.tsx

```diff
@@ -47,6 +52,52 @@ function WorkspaceContent({ workspaceId }: WorkspacePageProps) {
 
   const job = condenseJobs(workspace.jobs);
 
+  const datasetsSection =
+    workspaceDatasets.length > 0 ? (
+      <WorkspaceDatasetsTable
+        datasetsUUIDs={workspaceDatasets}
+        label={<SectionHeader> Datasets</SectionHeader>}
+        additionalButtons={
+          <WorkspacesUpdateButton
+            workspace={workspace}
+            dialogType="ADD_DATASETS"
+            sx={(theme) => ({
+              marginRight: theme.spacing(1),
+            })}
+            tooltip={tooltips.datasets}
+          >
+            <AddIcon />
+          </WorkspacesUpdateButton>
```

Comment:
Can we dedupe some of the shared components/logic between these two cases?

Reply:
We could just add a `boolean` prop to `WorkspaceDatasetsTable` to choose whether or not to show the copy button similar to how we handle `removeDatasets`. We could then display either the alert or the table based on whether the `datasetsUUIDs` passed to `WorkspaceDatasetsTable` is empty.

Reply:
I think this all makes sense, but I'd like to go over how I moved this logic into `WorkspaceDatasetsTable` - it alters how the empty table displays in other usages as well.